contracts/pallet-ibc/src/ics20_fee/mod.rs

```diff
@@ -262,10 +286,28 @@ where
 		// Send full amount to receiver using the default ics20 logic
 		// We only take the fee charge if the acknowledgement is not an error
 		if ack.as_ref() == Ics20Ack::success().to_string().as_bytes() {
-			// We have ensured that token amounts larger than the max value for a u128 are rejected
-			// in the ics20 on_recv_packet callback so we can multiply safely.
-			// Percent does Non-Overflowing multiplication so this is infallible
-			let fee = percent * packet_data.token.amount.as_u256().low_u128();
+			let asset_id =
+				<T as crate::Config>::IbcDenomToAssetIdConversion::from_denom_to_asset_id(
+					&packet_data.token.denom.to_string(),
+				);
+			let fee = match asset_id {
+				Ok(a) => {
+					let fee_asset_id = T::FlatFeeAssetId::get();
+					let fee_asset_amount = T::FlatFeeAmount::get();
+					let flat_fee =
+						T::FlatFeeConverter::get_flat_fee(a, fee_asset_id, fee_asset_amount)
+							.unwrap_or_else(|| {
+								// We have ensured that token amounts larger than the max value for
+								// a u128 are rejected in the ics20 on_recv_packet callback so we
+								// can multiply safely. Percent does Non-Overflowing multiplication
+								// so this is infallible
+								percent * packet_data.token.amount.as_u256().low_u128()
```

Comment:
we need to charge 10 USD in the value of that token. Is this what we are doing there?

Reply:
This is the case when there is no flat fee, so it charges the percentage of total